c7n/registry.py

```diff
@@ -48,6 +48,9 @@ def subscribe(self, func):
     def register(self, name, klass=None, condition=True,
                  condition_message="Missing dependency for {}",
                  aliases=None):
+
+        if '.' in name:
+            name = name.split('.', 1)[-1]
```

Comment:
Curious about the background for this.

I see that most of the `@resources.register()` calls in this PR have `aws.*` naming patterns (but not all, see `cloudwatch-dashboard` and `log-export`). These lines effectively ignore that provider prefix anyway though, so I'm wondering why it's there to begin with.

Reply:
some of this was generated, but Its hard to update the generated once its been added to the hand crafted files. I might try and go back and try to apply to a clean main with the generator updated.

Reply:
done, also fwiw experimenting with generating tests (using terraform doc examples for the resource).